setup.py

```diff
@@ -11,15 +11,15 @@
 requirements = [
     'dask[array] == 2021.2.0',
     'donfig >= 0.4.0',
-    'numpy >= 1.14.0',
+    'numpy >= 1.14.0, <= 1.19.5', # breakage in newer numpy + numerical errors
     'numba >= 0.43.0',
     'scipy >= 1.2.0',
     'threadpoolctl >= 1.0.0',
     'dask-ms == 0.2.6',
     'zarr >= 2.3.1'
 ]
 
-extras_require = {'testing': ['pytest', 'pytest-flake8', 'requests']}
+extras_require = {'testing': ['pytest', 'pytest-flake8', 'flake8 >= 4.0.0, <5.0.0', 'requests', 'gdown']}
```

Comment:
Pinning flake8 < 5 is due to https://github.com/tholo/pytest-flake8/issues/87#issuecomment-1201760236.

pre-commit hooks may be the way forward here, in future PRs

Reply:
Yea I'm in a bit of a fix all jenkins spree and I would like to get away with minimal changes in this PR. I will leave that till the next round where I can hopefully get to document and integrate the stuff @JSKenyon has been working on